lib/kubernetes-deploy/restart_task.rb

```diff
@@ -42,7 +42,7 @@ def initialize(context:, namespace:, logger: nil, max_watch_seconds: nil)
     def run(*args)
       perform!(*args)
       true
-    rescue FatalDeploymentError
+    rescue FatalDeploymentError, TaskConfigurationError
```

Comment:
Maybe we should also resue `DeploymentTimeoutError` since it raised https://github.com/Shopify/kubernetes-deploy/pull/583/files#diff-5068ffe6b6ef4e5064ed5138b4dae265L78 ?

Reply:
Technically you don't need to if you're going to do the same thing with the result, since `class DeploymentTimeoutError < FatalDeploymentError; end`

Reply:
Actually, so does `class TaskConfigurationError < FatalDeploymentError; end`. @dirceu 
 are you planning on changing that?

Reply:
Ah, good catch! I didn't notice that. No, I don't plan on changing that and I think that in that case this PR won't be needed.